test/agent_otel_merging/agent_otel_merging_test.go

```diff
@@ -0,0 +1,102 @@
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/util/common"
+)
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags()
+}
+
+// Tests merging of agent and OTEL yaml
+// Merges OTLP receiver to Agent yaml
+// And Checks if test_metric was generated(YAML merged successfully)
+func TestOtelMerging(t *testing.T) {
+	startAgent(t)
+	appendOtelConfig(t)
+	sendPayload(t)
+	verifyMetricsInCloudWatch(t)
+	verifyHealthCheck(t)
+}
+
+func startAgent(t *testing.T) {
+	common.CopyFile(filepath.Join("agent_configs", "config.json"), common.ConfigOutputPath)
+	require.NoError(t, common.StartAgent(common.ConfigOutputPath, true, false))
+	time.Sleep(10 * time.Second) // Wait for the agent to start properly
+}
+
+func appendOtelConfig(t *testing.T) {
+	cmd := exec.Command("sudo", "/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl",
+		"-a", "append-config", "-s", "-m", "ec2", "-c", "file:"+filepath.Join("resources", "otel.yaml"))
+	output, err := cmd.CombinedOutput()
+	require.NoError(t, err, "Failed to append OTEL config: %s", output)
+	time.Sleep(10 * time.Second) // Wait for the agent to apply the new configuration
+}
+
+func sendPayload(t *testing.T) {
+	currentTimestamp := time.Now().UnixNano()
+	payload, err := os.ReadFile(filepath.Join("resources", "metrics.json"))
+	require.NoError(t, err, "Failed to read JSON payload from config.json")
+	payloadStr := string(payload)
+	payloadStr = strings.ReplaceAll(payloadStr, "$CURRENT_TIMESTAMP", fmt.Sprintf("%d", currentTimestamp))
+
+	curlCmd := exec.Command("curl", "-X", "POST", "http://localhost:4318/v1/metrics",
+		"-H", "Content-Type: application/json", "-d", payloadStr)
+	curlOutput, err := curlCmd.CombinedOutput()
+	require.NoError(t, err, "Failed to send metrics: %s", curlOutput)
+	time.Sleep(1 * time.Minute)
+}
+
+func verifyMetricsInCloudWatch(t *testing.T) {
+	verifyCmd := exec.Command("aws", "cloudwatch", "get-metric-statistics",
+		"--namespace", "CWAgent-testing-otel",
+		"--metric-name", "test_metric",
+		"--dimensions", "Name=service.name,Value=test-service",
+		"--start-time", time.Now().Add(-5*time.Minute).Format("2006-01-02T15:04:05Z"),
+		"--end-time", time.Now().Format("2006-01-02T15:04:05Z"),
+		"--period", "60", // Period in seconds
+		"--statistics", "Average")
+
+	verifyOutput, err := verifyCmd.CombinedOutput()
+	require.NoError(t, err, "Failed to fetch metric from CloudWatch: %s", verifyOutput)
+
+	// Parse the CloudWatch response
+	type MetricStatistics struct {
+		Datapoints []struct {
+			Timestamp time.Time `json:"Timestamp"`
+			Average   float64   `json:"Average"`
+		} `json:"Datapoints"`
+		Label string `json:"Label"`
+	}
+
+	var stats MetricStatistics
+	err = json.Unmarshal(verifyOutput, &stats)
+	require.NoError(t, err, "Failed to parse CloudWatch response")
+	require.NotEmpty(t, stats.Datapoints, "No data points found for the metric")
+}
+
+func verifyHealthCheck(t *testing.T) {
```

Comment:
What's the purpose of this function? OTLP receiver does not use `13133` as default port iirc and it's not set to this port in the otel config either.

Reply:
So this is for the health check extension: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/healthcheckextension/README.md. Which listens on port 13133 this helps verify that extensions merge correctly in the yaml(one of the earlier issues we were having with yaml merging)